packages/expo-camera/ios/CameraPermissionsRequester.swift

```diff
@@ -0,0 +1,113 @@
+  func permissionWith(status systemStatus: AVAuthorizationStatus) -> [AnyHashable: Any]
+  func permissions(for key: String, mediaType: AVMediaType, service: String) -> [AnyHashable: Any]
+}
+
+extension BaseCameraRequester {
+  func permissions(for key: String, mediaType: AVMediaType, service: String) -> [AnyHashable: Any] {
+    var systemStatus: AVAuthorizationStatus
+    let description = Bundle.main.infoDictionary?[key] as? String
+
+    if let description {
+      systemStatus = AVCaptureDevice.authorizationStatus(for: mediaType)
+    } else {
+      EXFatal(EXErrorWithMessage("""
+      This app is missing \(key),
+      so \(service) services will fail. Add this entry to your bundle's Info.plist.
+      """))
+      systemStatus = .denied
+    }
+
+    return permissionWith(status: systemStatus)
+  }
+
+  func permissionWith(status systemStatus: AVAuthorizationStatus) -> [AnyHashable: Any] {
+    var status: EXPermissionStatus
+
+    switch systemStatus {
+    case .authorized:
+      status = EXPermissionStatusGranted
+    case .denied, .restricted:
+      status = EXPermissionStatusDenied
+    case .notDetermined:
+      fallthrough
+    @unknown default:
+      status = EXPermissionStatusUndetermined
+    }
+
+    return [
+      "status": status.rawValue
+    ]
+  }
+}
+
+class CameraOnlyPermissionRequester: NSObject, EXPermissionsRequester, BaseCameraRequester {
+  static func permissionType() -> String! {
+    "camera"
+  }
+
+  func getPermissions() -> [AnyHashable: Any] {
+    return permissions(for: cameraKey, mediaType: .video, service: "video")
+  }
+
+  func requestPermissions(resolver resolve: @escaping EXPromiseResolveBlock, rejecter reject: EXPromiseRejectBlock) {
+    AVCaptureDevice.requestAccess(for: .video) { [weak self] _ in
+      resolve(self?.getPermissions())
+    }
+  }
+}
+
+class CameraPermissionRequester: NSObject, EXPermissionsRequester, BaseCameraRequester {
+  static func permissionType() -> String {
+    "camera"
+  }
+
+  func getPermissions() -> [AnyHashable: Any] {
+    var systemStatus: AVAuthorizationStatus
+    var status: EXPermissionStatus
+
+    let cameraUsuageDescription = Bundle.main.infoDictionary?[cameraKey] as? String
+    let microphoneUsuageDescription = Bundle.main.infoDictionary?[microphoneKey] as? String
+
+    if let cameraUsuageDescription, let microphoneUsuageDescription {
+      systemStatus = AVCaptureDevice.authorizationStatus(for: .video)
+    } else {
+      EXFatal(EXErrorWithMessage("""
+      This app is missing either NSCameraUsageDescription or NSMicrophoneUsageDescription,
+      so audio/video services will fail. Add one of these entries to
+      your bundle's Info.plist
+      """))
+      systemStatus = .denied
+    }
+
+    return permissionWith(status: systemStatus)
+  }
+
+  func requestPermissions(resolver resolve: @escaping EXPromiseResolveBlock, rejecter reject: EXPromiseRejectBlock) {
+    AVCaptureDevice.requestAccess(for: .video) { [weak self] _ in
```

Comment:
This seems to be redundant as well. The only difference is whether you pass `.video` or `.audio`.

Reply:
Thought about removing it but it didn't seem worth it for a couple of lines. We still need a function that takes a media type and a closure so it ends up the same. Using a base class instead of a protocol doesn't improve it. We need empty implementations in the base class and can't override the static so need to add a `class` computed property. Ended up longer than the original code.

Reply:
Updated